components/ulp/ulp_riscv/ulp_core/ulp_riscv_print.c

```diff
@@ -49,3 +49,24 @@ void ulp_riscv_print_hex(int h)
         h <<= 4; // move the 2nd leftmost byte to the left, to be extracted next
     }
 }
+
+void ulp_riscv_print_hex_with_number_of_digits(int h, int number_of_digits)
+{
+    int x;
+    int c;
+
```

Comment:
Would be good to add a sanity check for `number_of_bytes < 0` which could return an error.